websocketOutportDriver/tests/realtest/client/client.go

```diff
@@ -0,0 +1,158 @@
+package client
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/core/mock"
+	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/gorilla/websocket"
+)
+
+// WSConn defines what a sender shall do
+type WSConn interface {
+	io.Closer
+	ReadMessage() (messageType int, p []byte, err error)
+	WriteMessage(messageType int, data []byte) error
+}
+
+var (
+	log              = &mock.LoggerMock{}
+	errNilMarshaller = errors.New("nil marshaller")
+)
+
+type tempClient struct {
+	name       string
+	marshaller marshal.Marshalizer
+	chanStop   chan bool
+}
+
+// NewTempClient will return a new instance of tempClient
+func NewTempClient(name string, marshaller marshal.Marshalizer) (*tempClient, error) {
+	if check.IfNil(marshaller) {
+		return nil, errNilMarshaller
+	}
+
+	return &tempClient{
+		name:       name,
+		marshaller: marshaller,
+		chanStop:   make(chan bool),
+	}, nil
+}
+
+// Run will start the client on the provided port
+func (tc *tempClient) Run(port int) {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	urlReceiveData := url.URL{Scheme: "ws", Host: fmt.Sprintf("127.0.0.1:%d", port), Path: "/operations"}
+	log.Info(tc.name+" -> connecting to", "url", urlReceiveData.String())
+	wsConnection, _, err := websocket.DefaultDialer.Dial(urlReceiveData.String(), nil)
+	if err != nil {
+		log.Error(tc.name+" -> dial", "error", err)
+	}
+	defer func() {
+		err = wsConnection.Close()
+		log.LogIfError(err)
+	}()
+
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+		for {
+			_, message, err := wsConnection.ReadMessage()
+			if err != nil {
+				log.Error(tc.name+" -> error read message", "error", err)
+				return
+			}
+
+			tc.verifyPayloadAndSendAckIfNeeded(message, wsConnection)
+		}
+	}()
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-done:
+			return
+		case t := <-ticker.C:
+			_ = t
+		case <-interrupt:
+			log.Info(tc.name + " -> interrupt")
+
+			// Cleanly close the connection by sending a close message and then
+			// waiting (with timeout) for the server to close the connection.
+			err := wsConnection.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
+			if err != nil {
+				log.Error(tc.name+" -> write close", "error", err)
+				return
+			}
+			select {
+			case <-done:
+			case <-time.After(time.Second):
+			}
+			return
+		}
+	}
+}
+
+func (tc *tempClient) verifyPayloadAndSendAckIfNeeded(payload []byte, ackHandler WSConn) {
+	if len(payload) == 0 {
+		log.Error(tc.name + " -> empty payload")
+		return
+	}
+
+	withAck := false
+	if payload[0] == byte(1) {
+		withAck = true
+	}
+
+	payload = payload[1:]
```

Comment:
by having this as a template client we will promote copying this payload processing over and over on all implementations. We need to have a wrapper client that does all this bytes operations and call an interface that all other clients will implement providing all the required information: payload, operation type and such. 
If we have this wrapper, we can automatically do the acknowledge handling without exposing it to the other implementation.
What we gain? We can easily change the protocol without changing a single line of code in all other clients and we will prevent code duplication such as L121-L127

Reply:
done, created a `payloadPerser` struct